{"priority": "trivial", "kind": "enhancement", "repository": {"links": {"self": {"href": "data/repositories/hoffmanlab/genomedata.json"}, "html": {"href": "#!/hoffmanlab/genomedata"}, "avatar": {"href": "data/bytebucket.org/ravatar/{78038659-1199-46ff-9d0a-675c62f89d60}ts=python"}}, "type": "repository", "name": "genomedata", "full_name": "hoffmanlab/genomedata", "uuid": "{78038659-1199-46ff-9d0a-675c62f89d60}"}, "links": {"attachments": {"href": "data/repositories/hoffmanlab/genomedata/issues/1/attachments_page=1.json"}, "self": {"href": "data/repositories/hoffmanlab/genomedata/issues/1.json"}, "watch": {"href": "https://api.bitbucket.org/2.0/repositories/hoffmanlab/genomedata/issues/1/watch"}, "comments": {"href": "data/repositories/hoffmanlab/genomedata/issues/1/comments_page=1.json"}, "html": {"href": "#!/hoffmanlab/genomedata/issues/1/no-sequence-leads-to"}, "vote": {"href": "https://api.bitbucket.org/2.0/repositories/hoffmanlab/genomedata/issues/1/vote"}}, "reporter": {"display_name": "Eric Roberts", "uuid": "{cd8c1fe0-28ca-45fb-8ef9-48090b42bb80}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7Bcd8c1fe0-28ca-45fb-8ef9-48090b42bb80%7D"}, "html": {"href": "https://bitbucket.org/%7Bcd8c1fe0-28ca-45fb-8ef9-48090b42bb80%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/17548d71dfb29d015b880f48cfc01ca3d=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsER-5.png"}}, "nickname": "ericr86", "type": "user", "account_id": "557058:c80ca578-03a1-4ac6-b3ee-50372a3fceee"}, "title": "No sequence leads to tables.exceptions.NoSuchNodeError instead of a more understandable error message", "component": null, "votes": 0, "watches": 2, "content": {"raw": "From Google Code Issue #18\r\n\r\n**Imported Labels**: bug, imported, Priority-Medium\r\n\r\n_From [epa...@uw.edu](https://code.google.com/u/108766868340988196614/) on September 12, 2012 19:03:07_\r\n\r\nIf I run the following python code\r\n\r\nimport genomedata\r\ngenome = genomedata.Genome(path_to_archive)\r\nchr1 = genome[\"chr1\"]\r\nchr1.seq[chr1.start:chr1.end]\r\n\r\nI get the error\r\n\r\n/net/noble/vol1/software/modules-sw/genomedata/1.3.2/Linux/RHEL5/x86_64/lib/python2.7/genomedata/__init__.py:1174: UserWarning: slice of chromosome sequence spans more than one supercontig (filling gaps with 'NaN')\r\n  warn(\"slice of chromosome sequence spans more than one supercontig\"\r\nTraceback (most recent call last):\r\n  File \"<stdin>\", line 1, in <module>\r\n  File \"/net/noble/vol1/software/modules-sw/genomedata/1.3.2/Linux/RHEL5/x86_64/lib/python2.7/genomedata/__init__.py\", line 1197, in __getitem__\r\n    seq[dest_start:dest_end] = supercontig.seq[sc_start:sc_end]\r\n  File \"/net/noble/vol1/software/modules-sw/genomedata/1.3.2/Linux/RHEL5/x86_64/lib/python2.7/genomedata/__init__.py\", line 1136, in seq\r\n    return self.h5group.seq\r\n  File \"/net/gs/vol3/software/modules-sw/python/2.7/Linux/RHEL5/x86_64/lib/python2.7/site-packages/tables/group.py\", line 834, in __getattr__\r\n    return self._f_getChild(name)\r\n  File \"/net/gs/vol3/software/modules-sw/python/2.7/Linux/RHEL5/x86_64/lib/python2.7/site-packages/tables/group.py\", line 705, in _f_getChild\r\n    self._g_checkHasChild(childname)\r\n  File \"/net/gs/vol3/software/modules-sw/python/2.7/Linux/RHEL5/x86_64/lib/python2.7/site-packages/tables/group.py\", line 446, in _g_checkHasChild\r\n    &#37; (self._v_pathname, name))\r\ntables.exceptions.NoSuchNodeError: group ``/supercontig_0`` does not have a child named ``seq``\r\n\r\nI am using Genomedata 1.3.2\r\n\r\n_Original issue: http://code.google.com/p/genomedata/issues/detail?id=18_", "markup": "markdown", "html": "<p>From Google Code Issue <a href=\"#!/hoffmanlab/genomedata/issues/18/assertions-caused-by-input-should-throw\" rel=\"nofollow\" title=\"Assertions caused by input should throw useful error messages\" class=\"ap-connect-link\">#18</a></p>\n<p><strong>Imported Labels</strong>: bug, imported, Priority-Medium</p>\n<p><em>From <a data-is-external-link=\"true\" href=\"https://code.google.com/u/108766868340988196614/\" rel=\"nofollow\">epa...@uw.edu</a> on September 12, 2012 19:03:07</em></p>\n<p>If I run the following python code</p>\n<p>import genomedata\ngenome = genomedata.Genome(path_to_archive)\nchr1 = genome[\"chr1\"]\nchr1.seq[chr1.start:chr1.end]</p>\n<p>I get the error</p>\n<p>/net/noble/vol1/software/modules-sw/genomedata/1.3.2/Linux/RHEL5/x86_64/lib/python2.7/genomedata/<strong>init</strong>.py:1174: UserWarning: slice of chromosome sequence spans more than one supercontig (filling gaps with 'NaN')\n  warn(\"slice of chromosome sequence spans more than one supercontig\"\nTraceback (most recent call last):\n  File \"&lt;stdin&gt;\", line 1, in &lt;module&gt;\n  File \"/net/noble/vol1/software/modules-sw/genomedata/1.3.2/Linux/RHEL5/x86_64/lib/python2.7/genomedata/<strong>init</strong>.py\", line 1197, in <strong>getitem</strong>\n    seq[dest_start:dest_end] = supercontig.seq[sc_start:sc_end]\n  File \"/net/noble/vol1/software/modules-sw/genomedata/1.3.2/Linux/RHEL5/x86_64/lib/python2.7/genomedata/<strong>init</strong>.py\", line 1136, in seq\n    return self.h5group.seq\n  File \"/net/gs/vol3/software/modules-sw/python/2.7/Linux/RHEL5/x86_64/lib/python2.7/site-packages/tables/group.py\", line 834, in <strong>getattr</strong>\n    return self._f_getChild(name)\n  File \"/net/gs/vol3/software/modules-sw/python/2.7/Linux/RHEL5/x86_64/lib/python2.7/site-packages/tables/group.py\", line 705, in _f_getChild\n    self._g_checkHasChild(childname)\n  File \"/net/gs/vol3/software/modules-sw/python/2.7/Linux/RHEL5/x86_64/lib/python2.7/site-packages/tables/group.py\", line 446, in _g_checkHasChild\n    &amp;<a href=\"#!/hoffmanlab/genomedata/issues/37/_load_datapy-python-script-interface-never\" rel=\"nofollow\" title=\"_load_data.py python script interface never used\" class=\"ap-connect-link\">#37</a>; (self._v_pathname, name))\ntables.exceptions.NoSuchNodeError: group <code>/supercontig_0</code> does not have a child named <code>seq</code></p>\n<p>I am using Genomedata 1.3.2</p>\n<p><em>Original issue: <a href=\"http://code.google.com/p/genomedata/issues/detail?id=18\" rel=\"nofollow\" class=\"ap-connect-link\">http://code.google.com/p/genomedata/issues/detail?id=18</a></em></p>", "type": "rendered"}, "assignee": {"display_name": "Eric Roberts", "uuid": "{cd8c1fe0-28ca-45fb-8ef9-48090b42bb80}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7Bcd8c1fe0-28ca-45fb-8ef9-48090b42bb80%7D"}, "html": {"href": "https://bitbucket.org/%7Bcd8c1fe0-28ca-45fb-8ef9-48090b42bb80%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/17548d71dfb29d015b880f48cfc01ca3d=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsER-5.png"}}, "nickname": "ericr86", "type": "user", "account_id": "557058:c80ca578-03a1-4ac6-b3ee-50372a3fceee"}, "state": "open", "version": null, "edited_on": null, "created_on": "2014-06-09T15:59:53.515733+00:00", "milestone": null, "updated_on": "2014-06-23T17:48:53.772477+00:00", "type": "issue", "id": 1}